src/main/scala/io/iohk/ethereum/ledger/BlockPreparator.scala

```diff
@@ -254,10 +255,18 @@ class BlockPreparator(
 
         validatedStx match {
           case Right(_) =>
-            val TxResult(newWorld, gasUsed, logs, _, _) = executeTransaction(stx, blockHeader, worldForTx)
+            val TxResult(newWorld, gasUsed, logs, _, vmError) = executeTransaction(stx, blockHeader, worldForTx)
+
+            // https://github.com/ethereum/EIPs/blob/master/EIPS/eip-658.md
+            val stateRoot = if (blockHeader.number >= blockchainConfig.byzantiumBlockNumber) {
+              val statusCode =  if (vmError.isDefined) 0 else 1
+              ByteString.fromInts(statusCode)
+            } else {
+              newWorld.stateRootHash
+            }
 
             val receipt = Receipt(
-              postTransactionStateHash = newWorld.stateRootHash,
+              postTransactionStateHash = stateRoot,
```

Comment:
hm i think it renaming would be confusing also, as on `etc` chain it will always be `postTransactionStateHash`, only on eth chain after byzantium block it will be indicator of success.
But definitly  some documentation in `Receipt.scala` about meaning of that field in both networks would be great , and some link to this eip.